### ISSUE TITLE
Error while connecting to URL

### ISSUE THREAD

**Issue by thylass** [theotherp/nzbhydra#426]
It is happening a lot lately with Dog. I'm aware of their daily maintenance, but most of the times it fails to connect in Hydra. 

`ERROR - search_module - Thread-222 - DogNZB: Error while connecting to URL https://api.dognzb.cr/api?apikey=<APIKEY>&t=tvsearch&extended=1&offset=0&limit=100&cat=5030,5040: HTTPSConnectionPool(host='api.dognzb.cr', port=443): Read timed out. (read timeout=30)`

The failure rate for Dog is 25% and the average response time is 14382ms. I highly doubt if has to do something with Hydra. Any suggestions and help are welcome and appreciated.

Thanks in advance!

**Comment by theotherp** [comment]
For me it's 14% failure rate and 3000ms response time. I have no idea how to analyze this issue. Would be interesting if other tools have similar problems.

**Comment by thylass** [comment]
I've changed the timeout to 50 seconds, and it's more like a hit and miss. It is the slowest indexer according to Hydra. I'm going to use it directly in Sonarr, and I will update the post shortly.

**Comment by thylass** [comment]
Yep, facing the same issue in Sonarr as well. 

`Dognzb.cr https://api.dognzb.cr/api?t=tvsearch&cat=5030,5040&extended=1&apikey=(removed)&offset=0&limit=100 Unable to connect to the remote server`

Guess it's Dog then.

**Comment by theotherp** [comment]
Contact their admin. There's a thread on reddit where one of their staff posted that they're aware of the issue and want to help.

**Comment by Spritzup** [comment]
I've been in contact with some of the guys who run Dog.  They're looking into the issue, though it may involve a rebuild of the API servers.

**Comment by seanvree** [comment]
Yeah, I'm having this issue as well! Hope there's a fix soon!

**Comment by theotherp** [comment]
So far I haven't seen any explanation why this happens. I don't have any issues connecting to Dog on my developer machine und on my dedicated server which are in different countries.

**Comment by seanvree** [comment]
@theotherp , right, it seems to be VERY intermittent.  But it does look like they are having problems on their side:

https://dognzb.cr/board/index.php?/topic/3403-is-api-down/

**Comment by combwizard** [comment]
Does it work at all for anyone? It's almost as if I'm getting blocked on purpose by DogNZB, because I haven't been able to get a single connection through to them using NZBHydra in a long, long time now. It's definitely not intermittent for me.

`2016-12-23 15:58:48,074 - INFO - newznab - Thread-1883 - Testing connection for host https://api.dognzb.cr
2016-12-23 15:59:09,164 - INFO - newznab - Thread-1883 - Unable to connect to indexer using URL https://api.dognzb.cr/api?t=tvsearch&apikey=<APIKEY>: HTTPSConnectionPool(host='api.dognzb.cr', port=443): Read timed out. (read timeout=20)`

**Comment by theotherp** [comment]
I noticed this happens when multiple requests are executed quickly. The first two or three work, the next ones all time out. Perhaps dog is using some sort of DDOS protection?

**Comment by combwizard** [comment]
Perhaps, but Sonarr isn't affected by this as far as I can tell. I can send dozens (even 100+) of queries to DogNZB in a short period using Sonarr without any protection kicking in. Heck, I can't even do a 'check capabilities' of the indexer without it timing out and the logs showing

`2016-12-30 14:34:49,249 - ERROR - newznab - Thread-85 - Error getting or parsing XML: HTTPSConnectionPool(host='api.dognzb.cr', port=443): Read timed out. (read timeout=20)`

I've checked and re-checked the settings probably 10 times to make sure I didn't enter my details wrong, and the IP of my NZBHydra is the same as my Sonarr, so I'm not IP banned, either.

But you know what, I changed the Agent to something other than NZBHydra, and it started returning searching. I can't be sure, but I think the admins at DogNZB just don't approve of NZBHydra.

**Comment by theotherp** [comment]
>  I can't be sure, but I think the admins at DogNZB just don't approve of NZBHydra and they are actively blocking it from accessing the API?

I don't think that's the case, they tried to help in the past. On the other side, their response can be kinda iffy... I haven't renewed my VIP status anyway.

**Comment by PhAzE-Variance** [comment]
Yea I get the same problem.  Hydra Log:
Did not pick DogNZB because it is disabled temporarily due to an error: Connection failed: HTTPSConnectionPool(host='api.dognzb.cr', port=443): Read timed out. (read timeout=20)

Trying to check for capabilities I get:  NoneType exception

**Comment by theotherp** [comment]
I don't know what to tell you guys. Dog has been so unreliable for me that I didn't renew my VIP.

**Comment by abhichandra21** [comment]
Which indexers do you use instead??
